client/my-sites/site-settings/section-import.jsx

```diff
@@ -16,31 +18,26 @@ import MediumImporter from 'my-sites/importer/importer-medium';
 import { fetchState } from 'lib/importer/actions';
 import { appStates, WORDPRESS, MEDIUM } from 'state/imports/constants';
 import EmailVerificationGate from 'components/email-verification/email-verification-gate';
+import { getSelectedSite } from 'state/ui/selectors';
 
-export default React.createClass( {
-	displayName: 'SiteSettingsImport',
+class SiteSettingsImport extends Component {
+	static propTypes = {
+		site: PropTypes.object,
+	};
 
-	propTypes: {
-		site: React.PropTypes.oneOfType( [
-			React.PropTypes.bool,
-			PropTypes.shape( {
-				slug: PropTypes.string.isRequired,
-				title: PropTypes.string.isRequired
-			} )
-		] )
-	},
+	constructor( props ) {
+		super( props );
 
-	componentDidMount: function() {
+		this.state = getImporterState();
```

Comment:
This makes complete sense @sirreal, and thanks for the catch.

To explain why I used the other approach: this is an old habit of my PHP background that setting dynamic data to class variables is not possible. Which is not true for JS of course 😉 

Fixed in https://github.com/Automattic/wp-calypso/pull/13182/commits/3f84e0e0955429d275cf2f1de2d25dfe5253586b.